src/visualisation/trigram.cc

```diff
@@ -212,6 +220,15 @@ int TrigramWidget::suggestBrightness() {
                   k_brightness_heuristic_max - offset);
 }
 
+VisualisationWidget::AdditionalResampleData* TrigramWidget::onAsyncResample() {
+  if (use_brightness_heuristic_) {
+    BrightnessData* res = new BrightnessData();
```

Comment:
It's later inserted into shared_ptr (base.cc, line 107), which will handle deletion.